internal/oneTBB/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+include(FetchContent)
+message(STATUS "Fetching oneTBB")
+set(TBB_TEST OFF CACHE BOOL "Build TBB Tests" FORCE)
+set(TBBMALLOC_BUILD OFF CACHE BOOL "Build tbbmalloc" FORCE)
+
+FetchContent_Declare(TBB
+        URL https://github.com/oneapi-src/oneTBB/archive/refs/tags/v2021.5.0.zip
+        URL_HASH SHA512=fde7eb56b46fa86b619eb60e587728d19f31639ec06dfd5d7af0dab00ffba38606d7a89dcf050c4cdfa8896560a7424a3431619108f395348dadf60ad6b67a1b
+
+        PREFIX "${CMAKE_CURRENT_BINARY_DIR}"
+        TMP_DIR "${CMAKE_CURRENT_BINARY_DIR}/tmp"
+        STAMP_DIR "${CMAKE_CURRENT_BINARY_DIR}/stamp"
+        DOWNLOAD_DIR "${CMAKE_CURRENT_BINARY_DIR}/dl"
+        SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/src"
+        SUBBUILD_DIR "${CMAKE_CURRENT_BINARY_DIR}/build"
+        BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}/bin"
+        INSTALL_DIR "${CMAKE_CURRENT_BINARY_DIR}/inst"
+        LOG_DIR "${CMAKE_CURRENT_BINARY_DIR}/log"
+
+        CONFIGURE_COMMAND "${CMAKE_COMMAND} -S ${CMAKE_CURRENT_BINARY_DIR}/src -B ${CMAKE_CURRENT_BINARY_DIR}/build"
+        INSTALL_COMMAND "${CMAKE_COMMAND} --install ${CMAKE_CURRENT_BINARY_DIR}/build --prefix \"${CMAKE_CURRENT_BINARY_DIR}/inst\""
+)
+FetchContent_Populate(TBB)
+
+# Manually change the output directory so folders don't spawn randomly in the build env
+set(TBB_OUTPUT_DIR_BASE "internal/oneTBB/build/" CACHE INTERNAL "Set output directory for oneTBB")
```

Comment:
Unfortunately even though we specify a SUBBUILD_DIR, TBB overrides the output directory for everything to `${CMAKE_BINARY_DIR}/${TBB_OUTPUT_DIR_BASE}_${_tbb_build_type_lower})` - since TBB is forcefully doing this without a way to override it, we just redirect it to include `internal/oneTBB/build` so that it will only create a `_${_tbb_build_type_lower}` folder inside and build there.

Reply:
Just for clarity's sake, an example - 
Using command `cmake ..` in `D:/repos/NovelRT/build`:

- Building NovelRT in `Debug` with MSVC (without this being set) outputs here: `D:/repos/NovelRT/build/msvc_19.32_cxx_64_md_debug`
- Building NovelRT in `Debug` with MSVC (_with this set as-is_) outputs here: `D:/repos/NovelRT/build/internal/oneTBB/build/_debug`

Did I also mention it sets this for libraries, archives, PDBs, and runtimes for both single and multi-config generators? :cringe:

Reply:
In that case, something like `file(RELATIVE_PATH location "${CMAKE_BINARY_DIR}" "${CMAKE_CURRENT_BINARY_DIR}")` would likely be better.

Reply:
This has been applied in the most recent commit